homeassistant/components/device_tracker/config_entry.py

```diff
@@ -37,9 +63,82 @@ async def async_unload_entry(hass: HomeAssistant, entry: ConfigEntry) -> bool:
     return await component.async_unload_entry(entry)
 
 
+@callback
+def _async_register_mac(
+    hass: HomeAssistant, domain: str, mac: str, unique_id: str | None = None
+) -> None:
+    """Register a mac address with a unique ID.
+
+    If no unique ID given, it is assumed to be the mac.
+    """
+    data_key = "device_tracker_mac"
+    mac = dr.format_mac(mac)
+    if data_key in hass.data:
+        hass.data[data_key][mac] = (domain, unique_id or mac)
+        return
+
+    # Setup listening.
+
+    # dict mapping mac -> partial unique ID
+    data = hass.data[data_key] = {mac: (domain, unique_id or mac)}
+
+    @callback
+    def handle_device_event(ev: Event) -> None:
+        """Enable the online status entity for the mac of a newly created device."""
+        # Only for new devices
+        if ev.data["action"] != "create":
+            return
+
+        dev_reg = dr.async_get(hass)
+        device_entry = dev_reg.async_get(ev.data["device_id"])
+
+        if device_entry is None:
+            return
+
+        # Check if device has a mac
+        mac = None
+        for conn in device_entry.connections:
+            if conn[0] == dr.CONNECTION_NETWORK_MAC:
+                mac = conn[1]
+                break
+
+        if mac is None:
+            return
+
+        # Check if we have an entity for this mac
+        if (unique_id := data.get(mac)) is None:
+            return
+
+        ent_reg = er.async_get(hass)
+        entity_id = ent_reg.async_get_entity_id(DOMAIN, *unique_id)
+
+        if entity_id is None:
+            return
+
+        entity_entry = ent_reg.async_get(entity_id)
+
+        if entity_entry is None:
+            return
+
+        # Make sure entity has a config entry and was disabled by the
+        # default disable logic in the integration.
+        if (
+            entity_entry.config_entry_id is None
+            or entity_entry.disabled_by != er.RegistryEntryDisabler.INTEGRATION
+        ):
+            return
+
+        # Enable entity
+        ent_reg.async_update_entity(entity_id, disabled_by=None)
+
+    hass.bus.async_listen(dr.EVENT_DEVICE_REGISTRY_UPDATED, handle_device_event)
+
+
 class BaseTrackerEntity(Entity):
     """Represent a tracked device."""
 
+    _attr_entity_category = EntityCategory.DIAGNOSTIC
```

Comment:
The block device info here as well (in typing that is)

```suggestion
    _attr_device_info: None = None
    _attr_entity_category = EntityCategory.DIAGNOSTIC
```